src/Simplex_tree/test/simplex_tree_edge_expansion_unit_test.cpp

```diff
@@ -0,0 +1,365 @@
+  }
+
+  return finalPoints;
+}
+
+template<class ST_type, typename St_options>
+void test_insert_as_flag(ST_type& simplex_tree, int maxDim = -1) {
+  Simplex_tree<St_options> st_to_test;
+  std::vector<typename Simplex_tree<St_options>::Simplex_handle> added_simplices;
+
+  for (auto& sh : simplex_tree.filtration_simplex_range()) {
+    if (simplex_tree.dimension(sh) == 0) {
+      auto u = *simplex_tree.simplex_vertex_range(sh).begin();
+      st_to_test.insert_edge_as_flag(u, u, simplex_tree.filtration(sh), maxDim, added_simplices);
+    } else if (simplex_tree.dimension(sh) == 1) {
+      auto it = simplex_tree.simplex_vertex_range(sh).begin();
+      auto u = *it;
+      auto v = *(++it);
+      st_to_test.insert_edge_as_flag(u, v, simplex_tree.filtration(sh), maxDim, added_simplices);
+    }
+  }
+
+  BOOST_CHECK(added_simplices.size() == simplex_tree.num_simplices());
+  BOOST_CHECK(st_to_test.dimension() == simplex_tree.dimension());
+  BOOST_CHECK(st_to_test == simplex_tree);
+}
+
+template <class ST_type, typename St_options>
+void test_unordered_insert_as_flag(ST_type& simplex_tree, int maxDim = -1) {
+  Simplex_tree<St_options> st_to_test;
+  std::vector<typename Simplex_tree<St_options>::Simplex_handle> added_simplices;
+
+  // complex_simplex_range gives a lexicographical order where the word "12" is shorter than "1",
+  // so vertices have to be inserted separately
+  for (auto& sh : simplex_tree.skeleton_simplex_range(0)) {
+    auto u = *simplex_tree.simplex_vertex_range(sh).begin();
+    st_to_test.insert_edge_as_flag(u, u, simplex_tree.filtration(sh), maxDim, added_simplices);
+  }
+
+  for (auto& sh : simplex_tree.complex_simplex_range()) {
+    if (simplex_tree.dimension(sh) == 1) {
+      auto it = simplex_tree.simplex_vertex_range(sh).begin();
+      auto u = *it;
+      auto v = *(++it);
+      st_to_test.insert_edge_as_flag(u, v, simplex_tree.filtration(sh), maxDim, added_simplices);
+    }
+  }
+  // as the insertions are not in the order of the filtration,
+  // the filtration values for higher dimensional simplices has to be restored
+  st_to_test.make_filtration_non_decreasing();
+
+  BOOST_CHECK(added_simplices.size() == simplex_tree.num_simplices());
+  BOOST_CHECK(st_to_test.dimension() == simplex_tree.dimension());
+  BOOST_CHECK(st_to_test == simplex_tree);
+}
+
+BOOST_AUTO_TEST_CASE(simplex_tree_random_rips_expansion) {
+  using Filtration_value = typename Simplex_tree<>::Filtration_value;
+  using Rips_complex = Gudhi::rips_complex::Rips_complex<Filtration_value>;
+
+  for (unsigned int i = 10; i < 100; i += 5){
+    Simplex_tree<> simplex_tree;
+    Rips_complex rips_complex(build_point_cloud(i, -1), 3, Gudhi::Euclidean_distance());
+    rips_complex.create_complex(simplex_tree, 100);
+
+    std::clog << i << "  ********************************************************************\n";
+    std::clog << "simplex_tree_random_rips_expansion\n";
+    std::clog << "Test tree 1: insertion by filtration values\n";
```

Comment:
I tried to run this test in debug mode, and I had some issues:
```txt
Running 3 test cases...
10  ********************************************************************
simplex_tree_random_rips_expansion
Test tree 1: insertion by filtration values
************************************************************************
unknown location(0): fatal error: in "simplex_tree_random_rips_expansion": std::invalid_argument: Simplex_tree::insert_edge_as_flag - inserts an already existing edge
```
It inserts correctly the verstices (0...9), but when it tries to insert the first edge [9, 4], it fails. I think the `GUDHI_CHECK(has_children(sh_u) && ..` is not true here.

Reply:
And I think the second part of the `GUDHI_CHECK( .. && sh_u->second.children()->members().find(v) != sh_u->second.children()->members().end()` is also not correct, because in the example, we insert the vertices (0...9), then the edges [7, 0], [8, 2], [6, 2] => Here the exception is also raised